src/dex/uniswap-v2/excalibur.ts

```diff
@@ -0,0 +1,62 @@
+import { UniswapV2 } from './uniswap-v2';
+import { Network } from '../../constants';
+import { Address, DexConfigMap } from '../../types';
+import { IDexHelper } from '../../dex-helper/index';
+import { Interface } from '@ethersproject/abi';
+import { DexParams } from './types';
+import excaliburPoolABI from '../../abi/uniswap-v2/excalibur-pool.json';
+import { getDexKeysWithNetwork } from '../../utils';
+
+export const ExcaliburConfig: DexConfigMap<DexParams> = {
+  Excalibur: {
+    [Network.FANTOM]: {
+      factoryAddress: '0x08b3CCa975a82cFA6f912E0eeDdE53A629770D3f',
+      initCode:
+        '0x3b43fe52e9f2b1864ca8a959ca3ac9c5fbc46f6379347e5f7d4e60b0ca479792',
+      feeCode: 0, // this is ingored as Excalibur uses dynamic fees
+    },
+  },
+};
+
+export class Excalibur extends UniswapV2 {
+  excaliburPool: Interface;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(ExcaliburConfig);
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper,
+  ) {
+    super(
+      network,
+      dexKey,
+      dexHelper,
+      true,
+      ExcaliburConfig[dexKey][network].factoryAddress,
+      ExcaliburConfig[dexKey][network].subgraphURL,
+      ExcaliburConfig[dexKey][network].initCode,
+      ExcaliburConfig[dexKey][network].feeCode,
+    );
+    this.feeFactor = 100000;
```

Comment:
Unfortunately, it's not valid to change feeFactor from 10000, because our smart contracts assume 10000.  Not sure why we left it possible to overwrite it like this.  The tests might still pass, but 10x fees will be charged on the swap in our router, which is obviously bad.  You will have to instead divide the fee by 10 (rounding up to next integer) in the callDecoder (prices will be marginally worse when fee is not a multiple of 10).  You can see in Biswap that we did the opposite to the fee (multiply by 10).